sources/api/host-containers/src/main.rs

```diff
@@ -135,7 +135,9 @@ mod error {
 type Result<T> = std::result::Result<T, error::Error>;
 
 /// Query the API for the currently defined host containers
-async fn get_host_containers<P>(socket_path: P) -> Result<HashMap<Identifier, model::HostContainer>>
+async fn get_host_containers<P>(
+    socket_path: P,
+) -> Result<HashMap<Identifier, settings_extension_host_containers::HostContainer>>
 where
     P: AsRef<Path>,
 {
```

Comment:
Below this we are calling API Server. In other other PRs, e.g. corndog, prariedog, et. al., we are eliminating reliance on the API Server by changing to a config. We will be doing that here as well right? Are we deciding to make such changes in two steps instead of one PR?

If we do, it raises the question of whether the config should be modeled as a map of `settings_extension_host_containers::HostContainer` structures. In other cases I don't think we have re-used the settings extension model directly in the config file.

Reply:
I was imagining that the setting extension and the config file change would be two PRs, I think they should be relatively separate changes. I can  take the config file work and get a PR out for that first.

> If we do, it raises the question of whether the config should be modeled as a map of `settings_extension_host_containers::HostContainer` structures. In other cases I don't think we have re-used the settings extension model directly in the config file.

The dogs should define their own structs to model the config files. We shouldn't depend on the settings model structs because those are no longer the source of truth for the setting shape, and we shouldn't depend on settings extensions because we can't guarantee that they'll be there

Reply:
2 PRs seems fine. If I understand correctly (and this is how I would want it): `host-containers` will no longer depend on the settings-extension once it has a config file, right?

Reply:
yep, that's what I was thinking